src/util/MusicSingleton.js

```diff
@@ -204,8 +204,9 @@ class MusicSingleton {
       const isInPlayingState =
         this.audioPlayer.state.status === AudioPlayerStatus.Playing;
       if (isInPlayingState) {
-        this.upcoming.push({ url, metadata: videoDetails });
-
+        for (let i = 0; i < repetitions; i++) {
+          this.upcoming.push({ url, metadata: videoDetails });
```

Comment:
if we want to repeat 100 times, how about we include the repetitions in the element, instead of adding 100 elements

so we could do something like
```suggestion
          this.upcoming.push({
            url,
            metadata: { ...videoDetails, repetitions },
          });
```

then when its time to pop the element from the list, we instead decrement the repetitions by 1. if the decremented value became zero, pop the element from the list